src/core/server/capabilities/capabilities_service.test.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { httpServiceMock, HttpServiceSetupMock } from '../http/http_service.mock';
+import { mockRouter, RouterMock } from '../http/router/router.mock';
+import { CapabilitiesService, CapabilitiesSetup } from './capabilities_service';
+import { mockCoreContext } from '../core_context.mock';
+
+describe('CapabilitiesService', () => {
+  let http: HttpServiceSetupMock;
+  let service: CapabilitiesService;
+  let setup: CapabilitiesSetup;
+  let router: RouterMock;
+
+  beforeEach(() => {
+    http = httpServiceMock.createSetupContract();
+    router = mockRouter.create();
+    http.createRouter.mockReturnValue(router);
+    service = new CapabilitiesService(mockCoreContext.create());
+  });
+
+  describe('#setup()', () => {
+    beforeEach(() => {
+      setup = service.setup({ http });
+    });
+
+    it('registers the capabilities routes', async () => {
+      expect(http.createRouter).toHaveBeenCalledWith('/api/core/capabilities');
+      expect(router.post).toHaveBeenCalledTimes(2);
+      expect(router.post).toHaveBeenCalledWith(expect.any(Object), expect.any(Function));
+    });
+
+    it('allows to register capabilities providers', async () => {
+      setup.registerProvider(() => ({
```

Comment:
could you add a couple of basic providers and switchers to make sure API works with arrays? `merge_capabilities` has similar tests, but we wanna be sure that we call mergeCapabilities with correct arguments